### DISCUSSION TITLE
Error in running nimbus-eth1 as execution client

### DISCUSSION THREAD

**Discussion by nidhi-singh02** [ethpandaops/ethereum-package#661] [Q&A]
I am running my network with multiple execution clients similar to how it can be done in this repository, where one EL client is nimbus-eth1. The nimbus-eth node is unable to come up. 

ERR failed to build block error="received nil payload ID on VALID engine response
ERR failed to prepare proposal err="received nil payload ID on VALID engine response" height=1 
ERR aborting block verification on nil block 
ERR failed to process proposal err="nil block" 
ERR prevote step: state machine rejected a proposed block; this should not happen:the proposer may be misbehaving; prevoting nil err=null height=1
INF rebuilding payload for rejected block 

What could be the issue here ?

**Comment by barnabasbusa** [comment]
I'd raise this issue with the nimbus-eth1 team, instead of here.

**Comment by nidhi-singh02** [comment]
Yeah, I get it. I have reached out to them already for the docker image for execution client, they said they have docker image except for testing. 
Today I saw in this repo, docker image being used from other entity `ethpandaops`, which the nimbus team does not maintain I believe. Hence raising it here as I am running it on kurtosis and using the same configuration for nimbus-eth1 as provided in this repository.

I would appreciate if you could help.

**Comment by advaita-saha** [comment]
@nidhi-singh02 
Nimbus-Eth1 is currently going through a database shift, from it's legacydb to Aristo, thus currently the engineAPI is not working as expected. We are on the process of fixing this issue

**Comment by nidhi-singh02** [comment]
Thanks @advaita-saha for the information. Any tentative dates by when it would be fixed ? Like when is the fix planned to be launched ?
One more confirmation: Is the docker image `ethpandaops/nimbus-eth1` maintained by the nimbus team ? I assume it is not.

**Comment by advaita-saha** [comment]
`ethpandaops/nimbus-eth1` is not maintained by the nimbus team.
Hopefully expect an alpha release in the coming months

**Reply by nidhi-singh02** [reply]
Thanks @advaita-saha , I was wondering how come `nimbus-eth1` execution client is working in this repository `ethereum-package` or is it breaking here as well @barnabasbusa.

**Comment by nidhi-singh02** [comment]
@advaita-saha I tried running the nimbus-eth1 image today, getting same results as earlier. Wanted to check with you if the DB migration  is done and engine API is fixed ?

**Reply by advaita-saha** [reply]
DB migration is done and work in engine API is going on

**Reply by nidhi-singh02** [reply]
Thanks for the update! Hopefully fixing the engine API will fix this.

**Comment by barnabasbusa** [comment]
Can you please paste your full yaml config that you are using?

**Reply by nidhi-singh02** [reply]
To simplify, currently I am using it in dockerized form - nimbus-eth1 as Execution layer client and my Consensus layer client is running as docker containers. 
I did start with the kurtosis package, seeing the basic docker containers not working as expected, shifted my focus.

**Reply by barnabasbusa** [reply]
It should be working. Thats why I'm asking what is the kurtosis config you used, so I can double check what is exactly not working.

**Reply by nidhi-singh02** [reply]
Sure, Here is the kurtosis config yaml file https://github.com/berachain/beacon-kit/pull/1323/files#diff-4fc364d810e87ce24d94023b2f3cd09c8ced17a4758fbd9063f21394258234fa
Would it be possible for you to share your Slack ID, I could create a channel for discussion. That would be really awesome.

**Reply by barnabasbusa** [reply]
Looks like there are quite some customizations to this script. 

Could you please provide a basic yaml file where you can replicate this bug, using only ethereum package specific configuration file.

Don't have any slack account but feel free to reach out to me on discord: @barnabasbusa

**Comment by barnabasbusa** [comment]
Should be sorted now. If not, please open an issue.
